opacus/privacy_engine/factory.py

```diff
@@ -0,0 +1,33 @@
+from opacus import PrivacyEngine
+from opacus.privacy_engine.catalog import (
+    PrivacyEngine,
+    PrivacyEngineNonPoisson,
+    PrivacyEnginePerLayer,
+    PrivacyEnginePerLayerNonPoisson,
+)
+
+
+class PrivacyEngineFactory:
+    @classmethod
+    def get(
+        cls,
+        accountant: str = "rdp",  # TODO: str or enum?
+        clipping: str = "flat",  # TODO: str or enum?
+        poisson_sampling: bool = True,
+        secure_mode: bool = False,
+    ):
+        if accountant != "rdp":
+            raise NotImplementedError
+
+        if clipping == "flat":
+            if poisson_sampling:
+                return PrivacyEngine(secure_mode)
+            else:
+                return PrivacyEngineNonPoisson(secure_mode)
```

Comment:
Very weak nit: is is better to have a single level switch instead of multi-level?
```suggestion
        if clipping == "flat" and poisson_sampling:
            return PrivacyEngine(secure_mode)
        elif clipping == "flat" and not poisson_sampling::
            return PrivacyEngineNonPoisson(secure_mode)
```